### PR TITLE
feat: Display source environment in LaunchDevly UI

### PR DESCRIPTION
**Pull request by cdelst**
**Requirements**

image

### CODE REVIEW

File: internal/dev_server/ui/src/Flags.tsx [hunk 1]
```diff
@@ -23,61 +29,70 @@ function Flags({ selectedProject, flags, setFlags }: FlagProps) {
     Record<string, FlagVariation[]>
   >({});
 
-  const overridesPresent = overrides && Object.keys(overrides).length > 0;
+  const overridesPresent = useMemo(
```
Comment (cdelst) [inline internal/dev_server/ui/src/Flags.tsx:32]: best practice to not recalculate these every time

File: internal/dev_server/ui/src/Flags.tsx [hunk 2]
```diff
@@ -23,61 +29,70 @@ function Flags({ selectedProject, flags, setFlags }: FlagProps) {
...
+    [overrides],
+  );
 
-  const updateOverride = (flagKey: string, overrideValue: LDFlagValue) => {
-    const updatedOverrides = {
-      ...overrides,
-      ...{
-        [flagKey]: {
-          value: overrideValue,
+  const updateOverride = useCallback(
```
Comment (cdelst) [inline internal/dev_server/ui/src/Flags.tsx:37]: useCallback lets us cache this func definition between renders

Comment (mike-zorn) [reply internal/dev_server/ui/src/Flags.tsx:37]: Do we want that though? We closure in the override state...

Comment (mike-zorn) [reply internal/dev_server/ui/src/Flags.tsx:37]: ohhh, I see you also pass in the deps.

**Review by mike-zorn — APPROVED** [review]

**Pull request merged**